src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php

```diff
@@ -0,0 +1,262 @@
+    {
+        $transitions = array();
+
+        foreach ($definition->getTransitions() as $name => $_) {
+            $transitions[$name] = array(
+                'attributes' => array('shape' => 'box', 'regular' => true),
+            );
+        }
+
+        return $transitions;
+    }
+
+    /**
+     * Returns all places.
+     *
+     * @return string A string representation of all places
+     */
+    private function addPlaces(array $places)
+    {
+        $code = '';
+
+        foreach ($places as $id => $place) {
+            $code .= sprintf("  place_%s [label=\"%s\", shape=circle%s];\n",
+                $this->dotize($id),
+                $id,
+                $this->addAttributes($place['attributes'])
+            );
+        }
+
+        return $code;
+    }
+
+    /**
+     * Returns all transitions.
+     *
+     * @return string A string representation of all transitions
+     */
+    private function addTransitions(array $transitions)
+    {
+        $code = '';
+
+        foreach ($transitions as $id => $place) {
+            $code .= sprintf("  transition_%s [label=\"%s\", shape=box%s];\n",
+                $this->dotize($id),
+                $id,
+                $this->addAttributes($place['attributes'])
+            );
+        }
+
+        return $code;
+    }
+
+    private function findEdges(Definition $definition)
+    {
+        $dotEdges = array();
+
+        foreach ($definition->getTransitions() as $transition) {
+            foreach ($transition->getFroms() as $from) {
+                $dotEdges[] = array(
+                    'from' => $from,
+                    'to' => $transition->getName(),
+                    'direction' => 'from',
+                );
+            }
+            foreach ($transition->getTos() as $to) {
+                $dotEdges[] = array(
+                    'from' => $transition->getName(),
+                    'to' => $to,
+                    'direction' => 'to',
+                );
+            }
+        }
+
+        return $dotEdges;
+    }
+
+    /**
+     * Returns all edges.
+     *
+     * @return string A string representation of all edges
+     */
+    private function addEdges($edges)
+    {
+        $code = '';
+
+        foreach ($edges as$edge) {
+            $code .= sprintf("  %s_%s -> %s_%s [style=\"solid\"];\n",
+                $edge['direction'] === 'from' ? 'place' : 'transition',
+                $this->dotize($edge['from']),
+                $edge['direction'] === 'from' ? 'transition' : 'place',
+                $this->dotize($edge['to'])
+            );
+        }
+
+        return $code;
+    }
+
+    /**
+     * Returns the start dot.
+     *
+     * @return string The string representation of a start dot
+     */
+    private function startDot(array $options)
+    {
+        return sprintf("digraph workflow {\n  %s\n  node [%s];\n  edge [%s];\n\n",
+            $this->addOptions($options['graph']),
+            $this->addOptions($options['node']),
```

Comment:
I fixed the constructor doc block. (`place` does not exist in the graphviz language.)